server/src/main/java/org/elasticsearch/cluster/metadata/IndexTemplateMetaData.java

```diff
@@ -281,13 +302,13 @@ public Builder settings(Settings settings) {
             return this;
         }
 
-        public Builder putMapping(String mappingType, CompressedXContent mappingSource) {
-            mappings.put(mappingType, mappingSource);
+        public Builder setMapping(CompressedXContent mappingSource) {
```

Comment:
here it is not very clear if we expect `mappingSource` to contain a type or not.  I saw examples of both